alternator/executor.cc

```diff
@@ -84,6 +84,21 @@ std::string make_jsonable::to_json() const {
     return rjson::print(_value);
 }
 
+json::json_return_type make_streamed(rjson::value&& value) {
+    // CMH. json::json_return_type uses std::function, not noncopyable_function. 
```

Comment:
What's "CMH"? Maybe say "Seastar's"?

Reply:
Interestingly, I wrote something very similar to this function the past, and had a long comment explaining the tradeoff between make_jsonable and the new make_streamed, and also when they both suck (e.g, BatchGetItem is forced to prepare the full json in advance).
Most importantly I mentioned (and even named the function accordingly, "make_jsonable_chunked") that one of the key differences is that the new function will cause the length of the response to not be known in advance, and chunked encoding will be used unconditionally. That's another reason to prefer not to use it for small responses.

Reply:
Also, please at least have a comment that explains that make_jsonable and make_streamed are two alternatives and what the differences are. It's not at all obvious from the name that make_streamed is an alternative to make_jsonable (BTW, I called my implementation make_jsonable_chunked to make that relation more obvious. 
By the way, I also re-implemented make_jsonable in the same way:
```c++
+struct value_jsonable : public json::jsonable {
+    rjson::value _value;
+public:
+    explicit value_jsonable(rjson::value&& value)
+        : _value(std::move(value))
+        {}
+    std::string to_json() const override {
+        return rjson::print(_value);
+    }
+};
+json::json_return_type make_jsonable(rjson::value&& value) {
+    return json::json_return_type(value_jsonable(std::move(value)));
+}
```

Reply:
CMH - "code modification here" - std::function is very bad for these kinds of callbacks due to copyable definition. noncopyable_function would be much better. 
I first made it an option to just send the object and assume we want streaming for this in server.cc, however, you drop a number of type aliasing "features" then, so it changes default behaviour potentially. 
Will add more comment.